### PR TITLE
#7 Board UI MockUp_Data

### PR DESCRIPTION
**Pull request by Vicheans**
### What this PR does (required):
- #7  Mock-up data for the Draggable FrontEnd [directory](https://github.com/hatchways/team-granite/blob/23_DragNDrop_to_Board/client/src/components/DragAndDrop/data.jsx)

### Screenshots / Videos (front-end only):
mockData
mockData0
mockData1
mockData2

### Any information needed to test this feature (required):
- @tejasj1718  @gami34 @moffatethan  Kindly take a look at the attached screenshots to see the suggested SCHEMA design for the Database. Kindly revert in case of changes on time for seamless integration. Many thanks!

### CODE REVIEW

File: client/src/components/DragAndDrop/dragNdrop.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const itemsFromBackend = [
+    {
+        boardKey: 3, tag: 'red', id: '1',
+        content: {
+            title: "Ronaldo Transfer To Man U", description: 'Complete This First',
+            deadline: new Date().toLocaleString(),
+            comments: [
+                { id: 1, text: 'Here We Go', user: 'Fabrizio Romano' }
+            ]
+        }
```
[the remark this answers is not in the document]
Comment (Vicheans) [reply client/src/components/DragAndDrop/dragNdrop.jsx]: Alright, Ethan. looking forward to the developments

File: client/src/components/DragAndDrop/dragNdrop.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+},
+];
+// const res = boards.reduce((board, val) => (board[val] = { name: val, items: [] }, board), {});
+//To DISSOLVE A BOARD OBJECT ARRAY INTO OBJECTS
+const columnsFromBackend = boardItems[0].boards.reduce((obj, board) => Object.assign(obj, { [board.key]: { name: board.name, items: board.items } }), {});
+
+const DragAndDrop = () =>{
+    return (<div>
+        <h1>Drag And Drop Component</h1>        
+    </div>)
```
[the remark this answers is not in the document]
Comment (Vicheans) [reply client/src/components/DragAndDrop/dragNdrop.jsx]: Alright, many thanks for the feedback.

**Comment by Vicheans** [comment]
> Left some comments on changing the mock data, as well as removing a unnecessary component. Thanks!

I got the component removal part, could you please point the correction for the mock data. Best regards

File: client/src/components/DragAndDrop/assets/item.module.scss [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  display: flex;
+  justify-content: center;
+  flex: auto;
+   & > div{
+    color: #fff;
+    padding: 2.5px;
+    border-radius: 10px;
+  }
+}
+
```
Comment (moffatethan) [inline client/src/components/DragAndDrop/assets/item.module.scss:24]: I actually would like to see you use the `makeStyles` from MUI as oppposed to Sass stylesheets.

File: client/src/components/DragAndDrop/drag&drop.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        if (combine) {
+            if (type === "COLUMN") {
+                const shallow = [...ordered].splice(source.index, 1);
+                setOrdered(shallow);
+                return;
+            }
+
+            const column = columns[source.droppableId];
+            const withItemRemoved = [...column].splice(source.index, 1);
+            const $_columns = {
```
Comment (moffatethan) [inline client/src/components/DragAndDrop/drag&drop.jsx:28]: Could you explain the naming choice here? I don't see the point of `$_columns`

File: client/src/components/DragAndDrop/drag&drop.jsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import classes from './assets/item.module.scss'
+import { Box, Grid } from '@material-ui/core';
+import { useBoardContext } from '../../context/useBoardContext';
+
+
+const Board = () => {
+
+    const { boardColumnMap: data, setBoardColumnMap } = useBoardContext();
+    const [columns, setColumns] = useState(data);
+    const [ordered, setOrdered] = useState(Object.keys(data))
```
Comment (moffatethan) [inline client/src/components/DragAndDrop/drag&drop.jsx:14]: Curious if we have to have a state for tracking the order? Couldn't we just work off the columns list and do the sorting there with array methods?

**Review by moffatethan — CHANGES_REQUESTED** [review]
Just some comments but overall this is coming together.

**Pull request merged**